lptree.pas

```diff
@@ -5759,6 +5897,63 @@ function TLapeTree_For.Compile(var Offset: Integer): TResVar;
   end;
 end;
 
+function TLapeTree_For.CompileForIn(var Offset: Integer): TResVar;
+var
+  arr, upper, lower: TResVar;
+  tmpExpr, method: TLapeTree_ExprBase;
+begin
+  Result := NullResVar;
+
+  if (not FLimit.CompileToTempVar(Offset, arr)) or (not arr.HasType()) then
+    LapeException(lpeInvalidEvaluation, FLimit.DocPos);
+
+  try
+    lower := _ResVar.New(FCompiler.getTempVar(ltInt32));
+    tmpExpr := TLapeTree_ResVar.Create(lower.IncLock(2), FCompiler);
+    method := TLapeTree_InternalMethod_Low.Create(tmpExpr);
+    TLapeTree_InternalMethod_Low(method).addParam(TLapeTree_ResVar.Create(arr.IncLock(), tmpExpr));
+    with TLapeTree_Operator.Create(op_Assign, Self) do
+    try
+      Left :=  TLapeTree_ResVar.Create(lower.IncLock(), FCompiler);
+      Right := method.FoldConstants() as TLapeTree_ExprBase;
+      Compile(Offset).Spill(1);
```

Comment:
I think you can remove Spill(1) here and use IncLock() instead of IncLock(2) a few lines up. The result of Compile op_Assign is Left (aka lower).